jina/flow/__init__.py

```diff
@@ -790,6 +794,54 @@ def my_reader():
         """
         self._get_client(**kwargs).search(input_fn, output_fn, **kwargs)
 
+    def flow_visualization(self):
+        """
+            Output the mermaid graph for visualization
+            :return: a mermaid-formatted string
+            """
+
+        mermaid_graph = list()
+
+        i = 0
+        for k in self._pod_nodes.items():
+            for need_pod in self._pod_needs[i]:
+                curr_line = "Pod" + need_pod + "[" + need_pod + "]" + "-->" + "Pod" + k[0] + "[" + k[0] + "]"
+                mermaid_graph.append(curr_line)
+            i = i+1
+
+        mermaid_str = 'graph TD\n' + '\n'.join(mermaid_graph)
+        return mermaid_str
+
+    def mermaidstr_to_url(self, **kwargs) -> str:
+        """
+        Rendering the current flow as a url points to a SVG, it needs internet connection
+        :param kwargs: keyword arguments of :py:meth:`to_mermaid`
+        :return: the url points to a SVG
+        """
+        import base64
+        mermaid_str = self.flow_visualization(**kwargs)
+        encoded_str = base64.b64encode(bytes(mermaid_str, 'utf-8')).decode('utf-8')
+
+        return 'https://mermaidjs.github.io/mermaid-live-editor/#/view/%s' % encoded_str
```

Comment:
isn't there a python library providing this functionality?

Reply:
Which functionality exactly you mean?